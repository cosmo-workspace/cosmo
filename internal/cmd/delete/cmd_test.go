package delete

import (
	"bytes"
	"testing"

	"github.com/cosmo-workspace/cosmo/pkg/cli"
	. "github.com/cosmo-workspace/cosmo/pkg/snap"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"

	"github.com/spf13/cobra"
)

func TestCommandDelete(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "cosmoctl delete suite")
}

var _ = Describe("help", func() {
	It("should match snapshot", func() {
		cmd := &cobra.Command{}
		out := bytes.Buffer{}
		cmd.SetOut(&out)
		AddCommand(cmd, cli.NewRootOptions())
		cmd.SetArgs([]string{"delete", "--help"})
		err := cmd.Execute()
		Expect(err).ShouldNot(HaveOccurred())
		Expect(out.String()).To(MatchSnapShot())
	})
})
