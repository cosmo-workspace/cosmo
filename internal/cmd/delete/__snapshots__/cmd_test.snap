['help should match snapshot 1']
SnapShot = """
Delete cosmo resources

Usage:
   delete [command]

Aliases:
  delete, rm, remove

Available Commands:
  network     Remove workspace network. Alias of 'cosmoctl workspace remove-network'
  user        Delete users. Alias of 'cosmoctl user delete'
  workspace   Delete workspaces. Alias of 'cosmoctl workspace delete'

Flags:
  -h, --help   help for delete

Use \" delete [command] --help\" for more information about a command.
"""
